Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.mm

```diff
@@ -91,20 +97,17 @@
     }
 
     if (prepareBuffersData.size()) {
-        auto& remoteRenderingBackend = layerTreeContext().ensureRemoteRenderingBackendProxy();
-        auto swapResult = remoteRenderingBackend.prepareBuffersForDisplay(WTFMove(prepareBuffersData));
-
-        RELEASE_ASSERT(swapResult.size() == backingStoreList.size());
-        for (unsigned i = 0; i < swapResult.size(); ++i) {
-            auto& backingStoreSwapResult = swapResult[i];
-            auto& backingStore = backingStoreList[i];
-            backingStore->applySwappedBuffers(WTFMove(backingStoreSwapResult.buffers.front), WTFMove(backingStoreSwapResult.buffers.back), WTFMove(backingStoreSwapResult.buffers.secondaryBack), backingStoreSwapResult.displayRequirement);
-        }
-    }
-
-    for (auto& backingStore : backingStoresWithNoBuffers) {
-        backingStore->applySwappedBuffers(nullptr, nullptr, nullptr, SwapBuffersDisplayRequirement::NeedsFullDisplay);
-        LOG_WITH_STREAM(RemoteLayerBuffers, stream << "RemoteLayerBackingStoreCollection::prepareBackingStoresForDisplay - allocated first buffers for layer " << backingStore->layer()->layerID() << " backing store " << *backingStore);
+        remoteRenderingBackend.prepareImageBufferSetsForDisplay(WTFMove(prepareBuffersData), [backingStoreList = WTFMove(backingStoreList)](Vector<RemoteRenderingBackendProxy::SwapBuffersResult>&& swapResult) {
+            RELEASE_ASSERT(swapResult.size() == backingStoreList.size());
```

Comment:
When the async call is cancelled, this will release assert.
It'd be better to handle the error intentionally.